conans/client/generators/compiler_args.py

```diff
@@ -57,9 +58,11 @@ def content(self):
             # Necessary in the "cl" invocation before specify the rest of linker flags
             flags.append(visual_linker_option_separator)
 
-        the_os = (self.conanfile.settings.get_safe("os_build") or
-                  self.conanfile.settings.get_safe("os"))
-        flags.extend(rpath_flags(the_os, self.compiler, self._deps_build_info.lib_paths))
+        os_build, _ = get_build_os_arch(self.conanfile)
+        if not hasattr(self.conanfile, 'settings_build'):
```

Comment:
This is also a bit complicated to read, you already checked the ``settings_build`` in the ``get_build_os_arch()`` call.

Reply:
totally agree, but here in the existing code we are using `settings.os` as a fallback. With the previous model, it could be a valid assumption that `settings.os` is a valid fallback-value for the _build context_, but if you provide a profile_build this is no longer a valid assumption.